pkg/networkservice/common/excludedprefixes/server.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2020 Cisco Systems, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package excludedprefixes provides a networkservice.NetworkServiceServer chain element that can read excluded prefixes
+// from config map and add them to request to avoid repeated usage.
+package excludedprefixes
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/trace"
+	"github.com/networkservicemesh/sdk/pkg/tools/prefixpool"
+)
+
+type excludedPrefixesService struct {
+	prefixes prefixpool.PrefixPool
+}
+
+func (eps *excludedPrefixesService) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	logger := trace.Log(ctx)
+
+	if request.GetConnection() == nil {
+		return nil, errors.Errorf("request's connection cannot be empty")
+	}
+
+	requestNext := request.Clone()
+	conn := requestNext.Connection
+	if conn.Context == nil {
```

Comment:
Context should not be nil, please comment this on NewServer documentation. And consider add a new chain element with making default values for endpoint and client default chains.

Reply:
ok, added comment